### ISSUE TITLE
False positive passing self to function in other file expecting object of class type

### ISSUE THREAD

**Issue by ethanbb** [DetachHead/basedpyright#1800]
### Description

I have code that looks like the following:

caiman_analysis.py:
```python
from . import foo
class SessionAnalysis:
    ...
    def print(self):
        foo.print_sessinfo(self)
```
foo.py:
```python
from . import caiman_analysis as cma

def print_sessinfo(x: 'cma.SessionAnalysis'):
    print(x)
```

The `self` argument to `print_sessinfo` is highlighted with the message:

```
Argument of type "Self@SessionAnalysis" cannot be assigned to parameter "x" of type "SessionAnalysis" in function "do_cnmf_gridsearch"
  "SessionAnalysis*" is not assignable to "SessionAnalysis"basedpyright[reportArgumentType](https://docs.basedpyright.com/v1.39.4/configuration/config-files/#reportArgumentType)
```

There's another instance where I'm passing `self` to another external function that expects an instance of this class, and I get the same error. Oddly, though, I don't get it when the function is in the same file (and annotated with `SessionAnalysis` or `'SessionAnalysis'` rather than `'cma.SessionAnalysis'`). Thus I tried to replicate it in the playground, but there's no error here: https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAEBiD28BcAoa7oBMCmAzaAHgBQTYi4CUqGN0ADgE4CWAdgC4D6uiJZlKKHPkatO3eEVxJoAcgTwZVNBhHtJFIA

I'm using version 1.39.4 in the VS code extension.

**Comment by DetachHead** [comment]
i can't reproduce this, but your example does have a circular dependency:

```
$ basedpyright
~/projects/test/src/caiman_analysis.py
  ~/projects/test/src/caiman_analysis.py: error: Cycle detected in import chain
    ~/projects/test/src/caiman_analysis.py
    ~/projects/test/src/foo.py (reportImportCycles)
1 error, 0 warnings, 0 notes
```

**Comment by ethanbb** [comment]
You're right, sorry, I realized that the error only shows up if I install the test package using `pip install -e . --config-settings editable_mode=compat`. This this the command I usually use to install editable packages; as noted in #731, it's needed for the local package to be recognized by basedpyright (and other type checkers) within other packages in the environment. This false positive doesn't show up if I don't use the `--config-settings` argument.

(I don't see the circular dependency error - probably disabled at 'basic' type checking level? - but this is not a problem for my use case.)

**Comment by DetachHead** [comment]
i still can't seem to reproduce even when installing with that command.

- does the same thing happen if you use uv instead of pip?
- what does your `pyproject.toml`/basedpyright config look like?
